### DISCUSSION TITLE
How to redirect to another url with the same prefix

### DISCUSSION THREAD

**Discussion by deeTEEcee** [chimurai/http-proxy-middleware#866] [General]
I'm a bit new to this so I might be a little off on wording. I have an application that currently uses something like:
```
module.exports = function(app) {
  app.use(
    createProxyMiddleware("/v1",{
      target: 'http://localhost:8090'
    })
  );
  app.use(
    createProxyMiddleware("/login",{
      target: 'http://localhost:8090'
    })
  );
```

This works for me but in our application, we ended up supporting something called "SCRIPT_PATH" to allow us to handle prefixes. On the backend side, my application now treats the following backend urls the same:
1. http://localhost/:endpoint:
2. http://localhost/connect/123/app/:endpoint:

where `:endpoint:` can be any continuation of that endpoint. Obviously, the first one still works but is it possible to get this working for the case where we've added in this new prefix?

**Comment by chimurai** [comment]
You might want to take a look at the `pathRewrite` option:
https://github.com/chimurai/http-proxy-middleware/blob/master/recipes/pathRewrite.md#pathrewrite

With the custom function I think you should be able to achieve want you want.
